### PR TITLE
fixes static gocilint errors

### PR DESCRIPTION
**Pull request by someshkoli**
ignores if false positive and adds err checkk where require

Signed-off-by: someshkoli <kolisomesh27@gmail.com>

### What problem does this PR solve?

Fixes gocilint static errs. Ignores where irrelevant.
Has some duplicate fixes from. #1773 , and #1779. Better merge those first then ill update so there's no duplications.
### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by STRRL** [comment]
Oh, that's a big PR and we need more time for reviewing.

And what's the usage of comment `// nolint`? It seems it has no relation with muse-dev. Could you give us any documents or references?

File: controllers/networkchaos/partition/types.go [hunk 1]
```diff
@@ -132,8 +132,14 @@ func (e *endpoint) Apply(ctx context.Context, req ctrl.Request, chaos v1alpha1.I
 			Namespace: pod.Namespace,
 		})
 
-		t.Append(sourceSet)
-		t.Append(targetSet)
+		e := t.Append(sourceSet)
```
Comment (AsterNighT) [inline controllers/networkchaos/partition/types.go:135]: pls use `err` instead of `e`

File: e2e-test/action.go [hunk 2]
```diff
@@ -113,7 +113,10 @@ func (oa *operatorAction) DeployOperator(info OperatorConfig) error {
 func (oa *operatorAction) InstallCRD(info OperatorConfig) error {
 	klog.Infof("deploying chaos-mesh crd :%v", info.ReleaseName)
 	oa.runKubectlOrDie("apply", "-f", oa.manifestPath("e2e/crd.yaml"), "--validate=false")
-	e2eutil.WaitForCRDsEstablished(oa.apiExtCli, labels.Everything())
+	er := e2eutil.WaitForCRDsEstablished(oa.apiExtCli, labels.Everything())
```
Comment (AsterNighT) [inline e2e-test/action.go:116]: ditto

File: e2e-test/e2e/chaos/networkchaos/network_delay.go [hunk 3]
```diff
@@ -100,6 +100,7 @@ func TestcaseNetworkDelay(
 	err := cli.Create(ctx, networkDelay.DeepCopy())
 	framework.ExpectNoError(err, "create network chaos error")
 
+	// nolint
```
Comment (AsterNighT) [inline e2e-test/e2e/chaos/networkchaos/network_delay.go:103]: I wonder what does this `// nolint` ignore? And what are the standards that you choose which error to fix and which to ignore?

File: pkg/chaosdaemon/client/utils.go [hunk 4]
```diff
@@ -20,7 +20,7 @@ import (
 	chaosdaemon "github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
 )
 
-var log = ctrl.Log.WithName("chaos-daemon-client-utils")
+var _ = ctrl.Log.WithName("chaos-daemon-client-utils")
```
Comment (AsterNighT) [inline pkg/chaosdaemon/client/utils.go:23]: If it is unused, you can just delete this line.

File: pkg/chaosdaemon/ipset_server.go [hunk 5]
```diff
@@ -51,7 +51,10 @@ func (s *DaemonServer) FlushIPSets(ctx context.Context, req *pb.IPSetsRequest) (
 		ipset := ipset
 		s.IPSetLocker.Lock(ipset.Name)
 		err := flushIPSet(ctx, req.EnterNS, pid, ipset)
-		s.IPSetLocker.Unlock(ipset.Name)
+		if err != nil {
```
Comment (AsterNighT) [inline pkg/chaosdaemon/ipset_server.go:54]: This is intended. Pls revert this.

File: pkg/chaosdaemon/server.go [hunk 6]
```diff
@@ -175,7 +175,9 @@ func StartServer(conf *Config, reg RegisterGatherer) error {
 		log.Info("Starting http endpoint", "address", httpBindAddr)
 		if err := httpServer.ListenAndServe(); err != nil {
 			log.Error(err, "failed to start http endpoint")
-			httpServer.Shutdown(context.Background())
+			if er := httpServer.Shutdown(context.Background()); err != nil {
+				return er
```
Comment (AsterNighT) [inline pkg/chaosdaemon/server.go:179]: use `err` instead of `er`

File: pkg/grpc/utils.go [hunk 7]
```diff
@@ -33,7 +33,7 @@ const DefaultRPCTimeout = 60 * time.Second
 // RPCTimeout specifies timeout of RPC between controller and chaos-operator
 var RPCTimeout = DefaultRPCTimeout
 
-var log = ctrl.Log.WithName("util")
+var _ = ctrl.Log.WithName("util")
```
Comment (AsterNighT) [inline pkg/grpc/utils.go:36]: ditto

**Review by AsterNighT — CHANGES_REQUESTED** [review]
FYI, your pr is fixing too many things and is conflict with almost every pr. You may have to fix lots of conflicts or break down your pr into several smaller ones..

**Comment by someshkoli** [comment]
> FYI, your pr is fixing too many things and is conflict with almost every pr. You may have to fix lots of conflicts or break down your pr into several smaller ones..

I agree, I should probably use this as a ref PR and raise small PR's and lets do this one at a time, WDYT ?

**Comment by AsterNighT** [comment]
> > FYI, your pr is fixing too many things and is conflict with almost every pr. You may have to fix lots of conflicts or break down your pr into several smaller ones..
> 
> I agree, I should probably use this as a ref PR and raise small PR's and lets do this one at a time, WDYT ?

Sry for the late reply. I suggest you check your fixes and pick out those which are already in master branch now first.

**Comment by STRRL** [comment]
It seems this PR will not going to be merged anymore, I am going to close it.

Feel free to reopen.

**Pull request closed without merging**